packages/nx/src/command-line/nx-commands.ts

```diff
@@ -36,6 +36,9 @@ export const commandsObject = yargs
     handler: async (args) => {
       // Remove the command from the args
       args._ = args._.slice(1);
+      process.env.NX_DRY_RUN = process.env.NX_DRY_RUN ?? `${args.dryRun}`;
```

Comment:
I think you can move these to a `.middleware()` call in `withGenerateOptions`. Also set `args.interactive` to `true` when `process.env.NX_INTERACTIVE !== 'false'`